File: 20210523LangReview/Python/07-enum/enum.cpp
#include <iostream>

enum class Color {
    red,
    green,
    blue
};

int main()
{
    Color r = Color::red;
    switch(r)
    {
        case Color::red  : std::cout << "red\n";   break;
        case Color::green: std::cout << "green\n"; break;
        case Color::blue : std::cout << "blue\n";  break;
    }

    int n = static_cast<int>(r);
    std::cout << n << '\n';
    
    return 0;
}

File: 20210523LangReview/Python/07-enum/enum.cs
using System;


namespace TS
{
    enum Color {
        red,
        green,
        blue
    }

    class TS
    {
        static void Main(string[] args)
        {
            Color r = Color.red;
            switch(r)
            {
                case Color.red:
                    print("red"); break;
                case Color.green:
                    print("green"); break;
                case Color.blue:
                    print("blue"); break;
            }

            Console.ReadLine();
        }
    }
}